wandb/sdk/interface/interface_shared.py

```diff
@@ -480,18 +520,55 @@ def _get_mailbox(self) -> Mailbox:
         assert mailbox
         return mailbox
 
-    def _deliver(self, record: pb.Record, slot_address: str) -> None:
-        record.control.mailbox_slot = slot_address
-        self._publish(record)
-
-    def _deliver_run(self, run: pb.RunRecord) -> MailboxHandle:
+    def _deliver_record(self, record: pb.Record) -> MailboxHandle:
         mailbox = self._get_mailbox()
-        rec = self._make_record(run=run)
-        handle = mailbox.get_handle()
-        slot_address = handle.address
-        self._deliver(rec, slot_address)
+        handle = mailbox._deliver_record(record, interface=self)
         return handle
 
+    def _deliver_run(self, run: pb.RunRecord) -> MailboxHandle:
+        record = self._make_record(run=run)
+        return self._deliver_record(record)
+
+    def _deliver_get_summary(self, get_summary: pb.GetSummaryRequest) -> MailboxHandle:
+        record = self._make_request(get_summary=get_summary)
+        return self._deliver_record(record)
+
+    def _deliver_exit(self, exit_data: pb.RunExitRecord) -> MailboxHandle:
+        record = self._make_record(exit=exit_data)
+        return self._deliver_record(record)
+
+    def _deliver_poll_exit(self, poll_exit: pb.PollExitRequest) -> MailboxHandle:
+        record = self._make_request(poll_exit=poll_exit)
+        return self._deliver_record(record)
+
+    def _deliver_request_server_info(
+        self, server_info: pb.ServerInfoRequest
+    ) -> MailboxHandle:
+        record = self._make_request(server_info=server_info)
+        return self._deliver_record(record)
+
+    def _deliver_request_sampled_history(
+        self, sampled_history: pb.SampledHistoryRequest
+    ) -> MailboxHandle:
+        record = self._make_request(sampled_history=sampled_history)
+        return self._deliver_record(record)
+
+    def _transport_keepalive_failed(self, keepalive_interval: int = 5) -> bool:
+        if self._transport_failed:
+            return True
+
+        now = time.time()
```

Comment:
does it make sense to use monotonic in this instance?

Reply:
yes... fixing this now.